### ISSUE TITLE
Reenable multiple calls to InitLogging

### ISSUE THREAD

**Issue by EvanBoyle** [pulumi/pulumi#4121]
Our recent change to use move to a multi-module repo https://github.com/pulumi/pulumi/pull/4109 caused our underlying logging library to get vendored in both `/pkg` and `/sdk`. At the time we were using `glog` which relies on global state initialized via the packages `func init()`. We moved to k8s.io/klog which avoids this by requiring explicit intialization through a function call. 

This method can only be called once, and is currently embedded in our 'InitLogging' function. This limits our usage to one init call.

**Comment by EvanBoyle** [comment]
the test fails with:

```
--- FAIL: TestInitLogging (0.00s)
panic: /var/folders/13/q5n976vd6vgc_wysnsmfkrbw0000gn/T/go-build125056891/b001/logging.test flag redefined: log_dir [recovered]
	panic: /var/folders/13/q5n976vd6vgc_wysnsmfkrbw0000gn/T/go-build125056891/b001/logging.test flag redefined: log_dir
```

**Comment by mikhailshilkov** [comment]
Fixed by #4167

**Comment by squaremo** [comment]
I'm unclear what is the status of this issue: on the one hand, "We moved to k8s.io/klog which avoids [relying on glog's global state]", and on the other, https://github.com/pulumi/pulumi/blob/d3e49ecb65832257da134740122d17be5c0ac88a/sdk/go/common/util/logging/log.go#L17, and the test mentioned now passes (e.g., https://github.com/pulumi/pulumi/runs/6659855666?check_suite_focus=true#step:37:63). Does that mean this is fixed?
